parser/parser.go

```diff
@@ -275,95 +285,11 @@ func (p *parser) parseReferences() {
 	for _, pkg := range p.pkgs {
```

Comment:
Could we use a call to `ast.Walk` in the future rather than the many nested loops?